tests/Mailer/MailerTest.php

```diff
@@ -0,0 +1,200 @@
+use Sonata\NewsBundle\Model\CommentInterface;
+use Sonata\NewsBundle\Model\PostInterface;
+use Sonata\NewsBundle\Util\HashGeneratorInterface;
+use Symfony\Component\Mailer\MailerInterface as SymfonyMailerInterface;
+use Symfony\Component\Mime\Email;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Templating\EngineInterface;
+
+class MailerTest extends TestCase
+{
+    /**
+     * @var RouterInterface|MockObject
+     */
+    private $router;
+
+    /**
+     * @var BlogInterface|MockObject
+     */
+    private $blog;
+
+    /**
+     * @var HashGeneratorInterface|MockObject
+     */
+    private $generator;
+
+    /**
+     * @var EngineInterface|MockObject
+     */
+    private $templating;
+
+    /**
+     * @var Mailer
+     */
+    private $mailer;
+
+    /**
+     * @var array
+     */
+    private $emails;
+
+    protected function setUp(): void
+    {
+        $this->mailer = $this->createMock(SymfonyMailerInterface::class);
+        $this->blog = $this->createMock(BlogInterface::class);
+        $this->generator = $this->createMock(HashGeneratorInterface::class);
+        $this->router = $this->createMock(RouterInterface::class);
+        $this->templating = $this->createMock(EngineInterface::class);
+        $this->emails = [
+            'notification' => [
+                'template' => 'email_test.html.twig',
+                'from' => 'news-bundle@sonata-project.org',
+                'emails' => ['news-bundle@sonata-project.org', 'news-bundle-anoher@sonata-project.org'],
+            ],
+        ];
+    }
+
+    /**
+     * NEXT_MAJOR: Remove this method.
+     *
+     * @group legacy
+     *
+     * @dataProvider emailTemplateData
+     */
+    public function testSendCommentNotificationWithSwiftMailer(string $template, string $subject, string $body): void
+    {
+        $post = $this->createMock(PostInterface::class);
+
+        $comment = $this->createStub(CommentInterface::class);
+        $comment
+            ->expects($this->once())
+            ->method('getPost')
+            ->willReturn($post);
+
+        $this->generator
+            //->expects($this->once())
+            ->method('generate')
+            ->with($comment)
+            ->willReturn('some_result');
+
+        $this->emails['notification']['template'] = $template;
+
+        $this->templating
+            ->expects($this->once())
+            ->method('render')
+            ->with($this->emails['notification']['template'], [
+                'comment' => $comment,
+                'post' => $post,
+                'hash' => 'some_result',
+                'blog' => $this->blog,
+            ])
+            ->willReturn($template);
+
+        $message = (new \Swift_Message())
+            ->setSubject($subject)
+            ->setFrom($this->emails['notification']['from'])
+            ->setTo($this->emails['notification']['emails'])
+            ->setBody($body);
+
+        $this->mailer = $this->createStub(\Swift_Mailer::class);
```

Comment:
Do we need to use `$this->mailer` here? Otherwise, I'd use a local variable instead, since the docblock at `MailerTest::$mailer` is only allowing instances of  `Sonata\NewsBundle\Mailer\Mailer`.

Reply:
It is require to change it in `getMailer()`. Local variable should be for SonataNewsBundle mailer.